CMakeLists.txt

```diff
@@ -72,8 +72,27 @@ file(GLOB_RECURSE SRC
         )
 
 #[[
-    Configure compilation target
+    Add googletest dependency
 ]]
+include(FetchContent)
+FetchContent_Declare(
+        googletest
+        URL https://github.com/google/googletest/archive/609281088cfefc76f9d0ce82e1ff6c30cc3591e5.zip
+)
+# For Windows: Prevent overriding the parent project's compiler/linker settings
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+FetchContent_MakeAvailable(googletest)
+
+# Gather all test files/related files needed for tests
+file(GLOB_RECURSE TESTS
+        "${CMAKE_SOURCE_DIR}/tests/*.cc"
+        "${CMAKE_SOURCE_DIR}/src/cmdhandler/command_handler.*"
+        )
+
+add_executable(AllTests ${PROTO_SRCS} ${PROTO_HDRS} ${TESTS})
+target_link_libraries(AllTests gtest gtest_main gmock ${OpenCV_LIBS} ${PROTOBUF_LIBRARIES} spdlog::spdlog stdc++fs)
```

Comment:
`stdc++fs` doesn't exist with MSVC. If you could remove it from here and either create a new `if` statement like the one below or just add `stdc++fs` within the existing `if(NOT MSVC)` statement for the `AllTests` target that would be 👌